api/v1alpha1/clusterfederatedtrustdomain_loader.go

```diff
@@ -33,34 +26,36 @@ func loadClusterFederatedTrustDomainFile(path string, scheme *runtime.Scheme, en
 
 	// Regardless of if the bytes are of any external version,
 	// it will be read successfully and converted into the internal version
-	if err = runtime.DecodeInto(codecs.UniversalDecoder(), content, entry); err != nil {
-		return fmt.Errorf("could not decode file (%s) into runtime.Object: %w", path, err)
+	if err = runtime.DecodeInto(codecs.UniversalDecoder(), content, &entry); err != nil {
+		return nil, fmt.Errorf("could not decode file (%s) into runtime.Object: %w", path, err)
 	}
 
-	return nil
+	return &entry, nil
 }
 
-func ListClusterFederatedTrustDomains(_ context.Context, scheme *runtime.Scheme, manifestPath string) ([]ClusterFederatedTrustDomain, error) {
+func ListClusterFederatedTrustDomains(_ context.Context, manifestPath string) ([]ClusterFederatedTrustDomain, error) {
+	scheme := runtime.NewScheme()
 	res := make([]ClusterFederatedTrustDomain, 0)
 	expandEnv := false
 	files, err := os.ReadDir(manifestPath)
 	if err != nil {
 		return nil, err
 	}
 	for _, file := range files {
-		var entry = ClusterFederatedTrustDomain{}
 		if !strings.HasSuffix(file.Name(), ".yaml") {
 			continue
 		}
 		fullfile := path.Join(manifestPath, file.Name())
-		err = LoadClusterFederatedTrustDomainFromFile(fullfile, scheme, &entry, expandEnv)
+		entry, err := loadClusterFederatedTrustDomainFile(fullfile, scheme, expandEnv)
+		// Ignore files of the wrong type in manifestPath
 		if entry.APIVersion != "spire.spiffe.io/v1alpha1" || entry.Kind != "ClusterFederatedTrustDomain" {
```

Comment:
is not this going to cause than all errors are avoid? because we are going to return an empty `entry` and continue is going to always call, 
may we log errors so users can understand why their files are not loaded? (same comment to statis entry)

Reply:
The same directory contains multiple types of files, parsed by different loops. so a file ignored here are picked up in say, api/v1alpha1/clusterstaticentry_loader.go. The only way to log the ignored ones would be to keep track of the files processed by each of the two reconciliation loops, synchronizing the passes, and then log the difference. Very tricky. :(

The juice is probably not worth the squeeze. Maybe we don't and if users request it, we reevaluate?